kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -0,0 +1,1234 @@
+        
+        // Check the values
+        if (available_options_for_constraints_scale.find(r_auxiliar_constraint_scale_factor) == available_options_for_constraints_scale.end()) {
+            std::stringstream msg;
+            msg << "Currently prescribed constraint scale factor : " << r_auxiliar_constraint_scale_factor << "\n";
+            msg << "Admissible values for the constraint scale factor are : use_mean_diagonal, use_diagonal_norm, or defined_in_process_info" << "\n";
+            KRATOS_ERROR << msg.str() << std::endl;
+        }
+        
+        // This case will consider the mean value in the diagonal as a scaling value
+        if (r_auxiliar_constraint_scale_factor == "use_mean_diagonal") {
+            mAuxiliarConstraintFactorConsidered = AUXILIAR_CONSTRAINT_FACTOR::CONSIDER_MEAN_DIAGONAL_CONSTRAINT_FACTOR;
+        } else if (r_auxiliar_constraint_scale_factor == "use_diagonal_norm") { // On this case the norm of the diagonal will be considered
+            mAuxiliarConstraintFactorConsidered = AUXILIAR_CONSTRAINT_FACTOR::CONSIDER_NORM_DIAGONAL_CONSTRAINT_FACTOR;
+        } else { // Otherwise we will assume we impose a numerical value
+            mAuxiliarConstraintFactorConsidered = AUXILIAR_CONSTRAINT_FACTOR::CONSIDER_PRESCRIBED_CONSTRAINT_FACTOR;
+        }
+        BaseType::mOptions.Set(BaseType::SILENT_WARNINGS, ThisParameters["silent_warnings"].GetBool());
+        if (ThisParameters["consider_lagrange_multiplier_constraint_resolution"].GetString() == "Double") {
+            BaseType::mOptions.Set(DOUBLE_LAGRANGE_MULTIPLIER, true);
+        } else {
+            BaseType::mOptions.Set(DOUBLE_LAGRANGE_MULTIPLIER, false);
+        }
+    }
+
+    /**
+     * @brief Default constructor.
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier(typename TLinearSolver::Pointer pNewLinearSystemSolver)   
+        : BaseType(pNewLinearSystemSolver)
+    {
+        // Setting flags
+        BaseType::mScalingDiagonal = BaseType::SCALING_DIAGONAL::NO_SCALING;
+        BaseType::mOptions.Set(BaseType::SILENT_WARNINGS, false);
+        mConstraintFactorConsidered = CONSTRAINT_FACTOR::CONSIDER_NORM_DIAGONAL_CONSTRAINT_FACTOR;
+        mAuxiliarConstraintFactorConsidered = AUXILIAR_CONSTRAINT_FACTOR::CONSIDER_NORM_DIAGONAL_CONSTRAINT_FACTOR;
+        BaseType::mOptions.Set(DOUBLE_LAGRANGE_MULTIPLIER, true);
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier() override
+    {
+    }
+
+    /**
+     * @brief Function to perform the build of the RHS. The vector could be sized as the total number
+     * of dofs or as the number of unrestrained ones
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rb The RHS vector
+     */
+    void Build(
```

Comment:
how is this different from the base class?

Reply:
It is adding the LM contributions

Reply:
i mean, here you call BaseType for everything relevant... it is identical to the baseclass